src/api.rs

```diff
@@ -1718,6 +1720,36 @@ impl Api {
         Ok(rv)
     }
 
+    /// Determines whether an organization exists or not. If the organization exists, returns
+    /// Err(ApiErrorKind::ProjectNotFound). If the organization does not exist, returns
+    /// Err(ApiErrorKind::OrganizationNotFound). If the user does not have permisisons to access
+    /// the organizations endpoint, we will be unable to determine whether the organization or
+    /// project doesn't exist, so we return Err(ApiErrorKind::OrganizationOrProjectNotFound).
+    /// If there is another error during the API request used to check the organization, that error
+    /// is returned.
+    ///
+    /// This function is intended to be called by functions where an API request was made,
+    /// which would fail identically if either the organization or project does not exist.
+    ///
+    /// This function always returns an Err() variant.
+    fn org_or_project_error<T>(&self, org: &str) -> ApiResult<T> {
+        let resp = self.get(&format!("/organizations/{}/", PathArg(org)))?;
+
+        if resp.ok() {
+            // The organization exists, so the project does not exist.
+            return Err(ApiErrorKind::ProjectNotFound.into());
+        }
+
+        match resp.status() {
+            404 => Err(ApiErrorKind::OrganizationNotFound.into()), // The organization does not exist.
+            403 => Err(ApiErrorKind::OrganizationOrProjectNotFound.into()), // The user does not have permissions to access the organizations endpoint.
```

Comment:
If the user doesn't have access to the endpoint, then do we _know_ that the org or project doesn't exist?

Reply:
Yes, if the user cannot access the organizations endpoint, then we know that the organization or the project does not exist (or both don't exist).

The calling code should only call this function if the user attempted to do something that would fail equivalently if the provided organization or the provided project does not exist. For example, if we receive a 404 error when querying the issues for a project, the URL will contain both the organization and the project, either an incorrect organization or an incorrect project could cause the 404. I documented this precondition for calling the function in the documentation comment on lines 1731-1732.

But, I was just thinking – perhaps it is simpler just to always return a `ProjectNotFound` error whenever a query like this fails. Since it is possible for an auth token to have permissions for querying issues but not for querying organizations, this function won't always be super helpful. And, even if the organization does not exist, it is still correct to say the project does not exist since a nonexistent org has no projects. We would just need to update the `ProjectNotFound` error message to inform users to check they entered the correct org as well

Reply:
We would also avoid the extra API request, and could even eliminate this new function. This change would become a three-line change if we switch to just always returning `ProjectNotFound`

Reply:
Yeah, that's a good point. Returning just `ProjectNotFound` with an expanded error message sounds simple and effective.